bugbot/rules/regression_set_status_flags.py

```diff
@@ -204,6 +207,45 @@ def get_status_changes(self, bugs):
                     info["esr"][f"esr{v}"] = "unaffected"
                 filtered_bugs[bugid] = info
 
+        # NOTE: The following is temporary to test that the new code is matching
+        # producing the expected results. Once we are confident with the new
+        # code, we can refactor this tool to use the new code.
+        # See: https://github.com/mozilla/bugbot/issues/2152
+        # >>>>> Begin of the temporary code
+        adjusted_bugs = (
+            {
+                **bug,
+                "regressed_by": [bug["regressed_by"]],
+            }
+            for bug in bugs.values()
+        )
+        bugs_store = BugsStore(
+            chain(adjusted_bugs, data.values()), utils.get_checked_versions()
+        )
+        for bug_id, old_code_updates in self.status_changes.items():
```

Comment:
This is only checking that old updates are matched by new. We are not checking if there are new ones that were not caught before

Reply:
Nice catch! Yes, this is true. Fixed in 765e1fe65a7b5f778cc9dcf840c4018455e1b493.